lib/p5.play.js

```diff
@@ -4338,4 +3909,1314 @@ p5.prototype._warn = function(message) {
+    var composedTransform = p5.Transform2D.mult(this._parentTransform, this._localTransform);
+    var transformedVertices = [
+      new p5.Vector(this._width / 2, -this._height / 2),
+      new p5.Vector(this._width / 2, this._height / 2),
+      new p5.Vector(-this._width / 2, this._height / 2)
+    ].map(function(vertex) {
+      return vertex.transform(composedTransform);
+    });
+
+    this._halfDiagonals = [
+      p5.Vector.sub(transformedVertices[0], this.center),
+      p5.Vector.sub(transformedVertices[1], this.center)
+    ];
+
+    this._potentialAxes = [
+      p5.Vector.sub(transformedVertices[1], transformedVertices[2]),
+      p5.Vector.sub(transformedVertices[1], transformedVertices[0])
+    ];
+  };
+
+  /**
+   * Debug-draw this collider.
+   * @method draw
+   * @param {p5} sketch - p5 instance to use for drawing
+   */
+  p5.OrientedBoundingBox.prototype.draw = function(sketch) {
+    var composedTransform = p5.Transform2D.mult(this._localTransform, this._parentTransform);
+    var scale = composedTransform.getScale();
+    var rotation = composedTransform.getRotation();
+    sketch.push();
+    sketch.translate(this.center.x, this.center.y);
+    sketch.scale(scale.x, scale.y);
+    if (sketch._angleMode === sketch.RADIANS) {
+      sketch.rotate(rotation);
+    } else {
+      sketch.rotate(degrees(rotation));
+    }
+
+    sketch.noFill();
+    sketch.stroke(0, 255, 0);
+    sketch.strokeWeight(1);
+    sketch.rectMode(sketch.CENTER);
+    sketch.rect(0, 0, this._width, this._height);
+    sketch.pop();
+  };
+
+  /**
+   * Compute candidate separating axes relative to another object.
+   * @method _getCandidateAxes
+   * @protected
+   * @return {Array.<p5.Vector>}
+   */
+  p5.OrientedBoundingBox.prototype._getCandidateAxes = function() {
+    // An oriented bounding box always provides two of its face normals,
+    // which we've precomputed.
+    return this._potentialAxes;
+  };
+
+  /**
+   * Get this shape's radius (half-width of its projection) along the given axis.
+   * @method _getRadiusOnAxis
+   * @protected
+   * @param {p5.Vector} axis
+   * @return {number}
+   */
+  p5.OrientedBoundingBox.prototype._getRadiusOnAxis = function(axis) {
+    // How to project a rect onto an axis:
+    // Project the center-corner vectors for two adjacent corners (cached here)
+    // onto the axis.  The larger magnitude of the two is your projection's radius.
+    return Math.max(
+      p5.Vector.project(this._halfDiagonals[0], axis).mag(),
+      p5.Vector.project(this._halfDiagonals[1], axis).mag());
+  };
+
+  /**
+   * A 2D affine transformation (translation, rotation, scale) stored as a
+   * 3x3 matrix that uses homogeneous coordinates.  Used to quickly transform
+   * points or vectors between reference frames.
+   * @class p5.Transform2D
+   * @constructor
+   * @extends Array
+   * @param {p5.Transform2D|Array.<number>} [source]
+   */
+  p5.Transform2D = function(source) {
```

Comment:
I actually got the collisions for rotated colliders all working and nice, but once I started trying to let you offset the colliders from their parent sprites I hit a wall.  That's when I realized I needed real transforms - and believe it or not, p5.js doesn't have a 2D transform matrix already.  So here is a 2D transform matrix class, using p5 conventions where I can and providing ways to modify the current matrix (to save constructing a new one) in many cases.